plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/form.tsx

```diff
@@ -150,6 +198,55 @@ const Form = < T extends AddressFormValues | ContactFormValues >( {
 					);
 				}
 
+				if ( field.key === 'address_2' ) {
+					return (
+						<Fragment key={ field.key }>
+							{ isAddress2FieldVisible ? (
+								<ValidatedTextInput
+									key={ field.key }
+									ref={ ( el ) =>
+										( fieldsRef.current[ field.key ] = el )
+									}
+									{ ...fieldProps }
+									type={ field.type }
+									value={ values[ field.key ] }
+									onChange={ ( newValue: string ) => {
+										onChange( {
+											...values,
+											[ field.key ]: newValue,
+										} );
+										setUserChangedAddress2Field( true );
+									} }
+									customValidation={ (
+										inputObject: HTMLInputElement
+									) =>
+										customValidationHandler(
+											inputObject,
+											field.key,
+											objectHasProp( values, 'country' )
+												? values.country
+												: ''
+										)
+									}
+								/>
+							) : (
+								<button
+									key={ field.key }
+									className={
+										'wc-block-components-address-form__address_2-toggle'
+									}
+									onClick={ toggleAddress2FieldVisibility }
+								>
+									{ __(
+										'+ Add apartment, suite, etc.',
```

Comment:
Can we keep the existing label? Without the `+`? Existing methods to change the label would stop working like this.

You can have the `+` in its own `_x` and replace `Add apartment, suite, etc` with label. That label will come with `(optional)` not sure if we want to keep that or not.

Reply:
This also limits locale options, because different countries might call this something else.

Reply:
> Can we keep the existing label? Without the +? Existing methods to change the label would stop working like this.

There's not really an existing label, as this is a new element, introduced by this PR. This text is for the link to open the `Apartment, suite, etc.` input field. I could use the existing label of the input field, but that label might not be speaking enough. Here's what I mean:

<table>
<tr>
<td valign="top">Default label of input field:
<br><br>
<img width="725" alt="Screenshot 2024-05-13 at 17 47 49" src="https://github.com/woocommerce/woocommerce/assets/3323310/ac555202-ddf4-4d07-9440-79ab4fac94c4">
</td>
<td valign="top">Custom label for the link:
<br><br>
<img width="705" alt="Screenshot 2024-05-13 at 17 48 48" src="https://github.com/woocommerce/woocommerce/assets/3323310/7dab8e19-c514-4ee7-8f01-875261c01776">
</td>
</tr>
</table>

I have two potential options in mind to avoid having the `+` sign in the translation:

1. We introduce two `<span>` elements. One for the `+` sign, one for the translation, and add some spacing behind the `+` sign using CSS:
```jsx
<button
	key={ field.key }
	className={
		'wc-block-components-address-form__address_2-toggle'
	}
	onClick={ toggleAddress2FieldVisibility }
>
	<span id="wc-block-components-address-form__address_2-toggle-icon">
		+
	</span>
	<span id="wc-block-components-address-form__address_2-toggle-label">
		{ __( 'Add apartment, suite, etc.', 'woocommerce' ) }
	</span>
</button>
```
```css
#wc-block-components-address-form__address_2-toggle-icon {
	margin-right: $gap-smallest;
}
```
2. We add the `+` sign via CSS:
```jsx
<button
	key={ field.key }
	className={
		'wc-block-components-address-form__address_2-toggle'
	}
	onClick={ toggleAddress2FieldVisibility }
>
	{ __( 'Add apartment, suite, etc.', 'woocommerce' ) }
</button>

```

```css
.wc-block-components-address-form__address_2-toggle {
	[...]

	&::before {
		content: '+ ';
	}
}
```

I favour the 2nd option, as it does not add extra markup to the site and allows merchants to overwrite the `+` sign much easier. What are your thoughts on these options? If you have another option in mind, I'd been keen in hearing it.

Reply:
I don't see the need to have two elements, simply doing `Add %s` with a comment is enough here.

Reply:
Good call. I've implemented this in https://github.com/woocommerce/woocommerce/pull/47160/commits/ebca77947c0208816d108a8a391298f76a06462a.